src/game/Spells/Spell.cpp

```diff
@@ -2627,17 +2629,19 @@ void Spell::SetTargetMap(SpellEffectIndex effIndex, uint32 targetMode, UnitList&
 
                 if (i_spellST->second.type == SPELL_TARGET_TYPE_GAMEOBJECT)
                 {
-                    // search all GO's with entry, within range of m_destN
-                    MaNGOS::GameObjectEntryInPosRangeCheck go_check(*m_caster, i_spellST->second.targetEntry, x, y, z, radius);
-                    MaNGOS::GameObjectListSearcher<MaNGOS::GameObjectEntryInPosRangeCheck> checker(tempTargetGOList, go_check);
-                    Cell::VisitGridObjects(m_caster, checker, radius);
+                    // Search all GO's with entry, within range of m_destN. Pick nearest first.
+                    GetGameObjectListWithEntryInGrid(tempTargetGOList, m_casterUnit, i_spellST->second.targetEntry, radius);
+                    tempTargetGOList.sort(ObjectDistanceOrder(m_casterUnit));
```

Comment:
wtf are these changes to spell target code? it is supposed to search around the x, y, z which are set above according to target type
you are also assuming caster is always unit by using m_casterUnit, which can crash server